sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -2202,6 +2202,36 @@ class DataFrameAggregateSuite extends QueryTest
     assertAggregateOnDataframe(dfDifferent, numRows, "m0")
   }
 
+  test("Fix reference error when grouping by MapType") {
```

Comment:
Let's have some tests for
- multiple group-by columns
- multiple maptype group-by columns
- maptype in aggregate expressions
- maptype in non-aggregate expression, but in another expression
- selecting other rows
to confirm behavior and non breaking

Reply:
Added a couple of more tests.